xarray/tests/test_backends.py

```diff
@@ -1267,6 +1267,133 @@ def test_4_open_large_num_files_h5netcdf(self):
+
+        ds1 = Dataset(data_vars={self.var_name: (['t', 'x'], v1),
+                                 self.coord_name: ('x', 2 * x)},
+                      coords={
+                          't': (['t', ], t1),
+                          'x': (['x', ], x)
+                      })
+
+        ds2 = Dataset(data_vars={self.var_name: (['t', 'x'], v2),
+                                 self.coord_name: ('x', 2 * x)},
+                      coords={
+                          't': (['t', ], t2),
+                          'x': (['x', ], x)
+                      })
+
+        return ds1, ds2
+
+    def test_open_mfdataset_does_same_as_concat(self):
+        with create_tmp_file() as tmpfile1:
+            with create_tmp_file() as tmpfile2:
+                ds1, ds2 = self.gen_datasets_with_common_coord_and_time()
+
+                # save data to the temporary files
+                ds1.to_netcdf(tmpfile1)
+                ds2.to_netcdf(tmpfile2)
+
+                files = [tmpfile1, tmpfile2]
+                for opt in ['all', 'minimal']:
+                    with open_mfdataset(files, data_vars=opt) as ds:
+                        kwargs = dict(data_vars=opt, dim='t')
+                        ds_expect = xr.concat([ds1, ds2], **kwargs)
+
+                        data = ds[self.var_name][:]
+                        data_expect = ds_expect[self.var_name][:]
+
+                        coord = ds[self.coord_name][:]
+                        coord_expect = ds_expect[self.coord_name][:]
+
+                        self.assertArrayEqual(data, data_expect)
+                        self.assertArrayEqual(coord, coord_expect)
+
+    def test_common_coord_dims_should_change_when_datavars_all(self):
+        with create_tmp_file() as tmpfile1:
+            with create_tmp_file() as tmpfile2:
+                ds1, ds2 = self.gen_datasets_with_common_coord_and_time()
+
+                # save data to the temporary files
+                ds1.to_netcdf(tmpfile1)
+                ds2.to_netcdf(tmpfile2)
+
+                files = [tmpfile1, tmpfile2]
+                # open the files with the default data_vars='all'
+                with open_mfdataset(files, data_vars='all') as ds:
+
+                    coord_shape = ds[self.coord_name].shape
+                    coord_shape1 = ds1[self.coord_name].shape
+                    coord_shape2 = ds2[self.coord_name].shape
+
+                    var_shape = ds[self.var_name].shape
+                    var_shape1 = ds1[self.var_name].shape
+                    var_shape2 = ds2[self.var_name].shape
+
+                    self.assertNotEqual(coord_shape1, coord_shape)
+                    self.assertNotEqual(coord_shape2, coord_shape)
+
+                    self.assertEqual(var_shape[0],
+                                     var_shape1[0] + var_shape2[0])
+
+                    self.assertEqual(var_shape, coord_shape)
+
+    def test_common_coord_dims_should_not_change_when_datavars_minimal(self):
```

Comment:
This looks very similar to the last test -- can you maybe consolidate it?

Or you could even potentially drop some of these tests. We have unit tests for `concat` and `open_mfdataset` already, so the main thing we need to verify is that the keyword argument gets properly passed on. We don't need to check here that every possible way to use it is handled correctly.